python/ray/air/execution/_internal/actor_manager.py

```diff
@@ -401,6 +401,8 @@ def on_error(exception: Exception):
 
                 self._enqueue_cached_actor_tasks(tracked_actor=tracked_actor)
 
+                started_actors += 1
```

Comment:
At the beginning, I am surprised that this didn't cause errors before. But after some investigation, seems that the returned `started_actors` is not being used in any place. Do we still need it as a return value?

Reply:
The return value is not used anywhere, but the local variable is still used to [break out of the loop of attempting to schedule actors.](https://github.com/ray-project/ray/pull/42951/files/7875918a3d874cd81f676180f54b922cdafad38d#diff-78a2208d861efe3970b8257d3b598d708898d871e6e757884cb3c76cfdaf2e61R327)

I think this may not be a huge issue because actor scheduling is still limited by the resources available in the cluster. This `max_actors` is set to 1 everywhere, so the fact that this counter wasn't working means we have been processing more than one actor scheduling request in one iteration of the event loop.

But that is probably ok:

```python
for i in range(1):
    # schedule 5 actors

# vs.

for i in range(5):
    # schedule 1 actor
```

Reply:
Got it. Previously this function could exhaust the cluster resource and try to launch as many actors as possible. We should re-enable the restriction of `max_actors`.